### PR TITLE
Glowcap spread endurance loss is 10 times bigger

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
Adds a hardlimit to how much glowcap can spread(5).If maintainers demand it i can make it based off potency or a config option.
## Why It's Good For The Game
Stops glowcaps from overtaking the station , reduces subsequent lag that comes from all the lightning that has to be calculated.

## Changelog
:cl:
tweak: Glowcaps now lose 10 times more endurance due to spread.
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -38. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by coiax** [comment]
> config option

Oh no

**Comment by interestingusernam3** [comment]
tweak

**Comment by Krysonism** [comment]
There is already a cap on how many generations can spread, it is called endurance.

A better solution would have glowshrooms lose endurance faster.

**Comment by interestingusernam3** [comment]
How the hell do you have -38 GBP at the time of your first contribution to the codebase ever?

**Comment by MLGTASTICa** [comment]
> How the hell do you have -38 GBP at the time of your first contribution to the codebase ever?

Contributed on TGMC.

**Comment by MLGTASTICa** [comment]
> There is already a cap on how many generations can spread, it is called endurance.
> 
> A better solution would have glowshrooms lose endurance faster.

Yes , but on a downstream it has happened quite frequently that glowshrooms go over 10+ generations which is not fine for performance . Im also assuming that the botany interactions are still being called on the glowshrooms which makes them recover potency/yield .  They also have a chance to receive more yield which makes them roll more times for spreading.

**Comment by NotRanged** [comment]
Make it scale with stats like endurance/production/potency and add some logging to INVESTIGATE_BOTANY about planting them perhaps

**Comment by Jared-Fogle** [comment]
> If maintainers demand it i can make it based off potency or **a config option**.

you're a card 😊

File: code/game/objects/effects/glowshroom.dm [hunk 1]
```diff
@@ -112,6 +112,9 @@
 	//We could be deleted at any point and the timers might not be cleaned up
 	if(QDELETED(src))
 		return
+	// If its at the third generation we want it to stop spreading.
+	if(generation>2)
```
Comment (Jared-Fogle) [inline code/game/objects/effects/glowshroom.dm:116]: Magic number, move it to a define.

Comment (MLGTASTICa) [reply code/game/objects/effects/glowshroom.dm]: Moved it to a define in botany defines.

**Comment by Mothblocks** [comment]
what @Krysonism said though

**Review by Mothblocks — CHANGES_REQUESTED** [review]
There is already a cap on how many generations can spread, it is called endurance.

A better solution would have glowshrooms lose endurance faster.

**Comment by MLGTASTICa** [comment]
> There is already a cap on how many generations can spread, it is called endurance.
> 
> A better solution would have glowshrooms lose endurance faster.

Done , they now lose 10 times more endurance.

File: code/game/objects/effects/glowshroom.dm [hunk 2]
```diff
@@ -171,7 +168,7 @@
 			if(shroomCount >= placeCount)
 				continue
 
-			Decay(TRUE, 2) // Decay before spawning new mushrooms to reduce their endurance
+			Decay(TRUE, 20) // Decay before spawning new mushrooms to reduce their endurance
```
Comment (Mothblocks) [inline code/game/objects/effects/glowshroom.dm:171]: Move to define, then it's good.

Comment (MLGTASTICa) [reply code/game/objects/effects/glowshroom.dm]: Done

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-46). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**